api/src/main/java/org/openmrs/CohortMembership.java

```diff
@@ -99,6 +118,20 @@ public void setEndDate(Date endDate) {
 
 	@Override
 	public int compareTo(CohortMembership o) {
-		return this.getPatient().getPatientId() - o.getPatient().getPatientId();
+		if (Objects.equals(this.getPatient().getPatientId(), o.getPatient().getPatientId())
+				&& Objects.equals(this.getCohort().getCohortId(), o.getCohort().getCohortId())
+				&& this.getStartDate().equals(o.getStartDate())
+				&& ((this.getEndDate() != null && o.getEndDate() != null
+				&& this.getEndDate().equals(o.getEndDate()))
+				|| (this.getEndDate() == null && o.getEndDate() == null))) {
+			return 0;
+		} else if ((this.isMembershipActive() && !o.isMembershipActive())
+				|| (this.startDate.before(o.startDate))) {
+			return -1;
+		} else if ((!this.isMembershipActive() && o.isMembershipActive())
+				|| (this.startDate.after(o.startDate))) {
+			return 1;
+		}
+		return -1;
```

Comment:
`return OpenmrsUtil.compare(Date, Date)` seems to be breaking some tests in CohortTest. 

<img width="572" alt="screen shot 2017-04-14 at 6 05 29 pm" src="https://cloud.githubusercontent.com/assets/193819/25059567/abf17190-213d-11e7-89e3-df328112dd4e.png">

Assert of `cohort.contains(id)` fails. It can no longer match patient ids. Maybe because the ids are different and the start dates match?